### DISCUSSION TITLE
Downloading sample C2M2 TSVs

### DISCUSSION THREAD

**Discussion by jeremyjyang** [nih-cfde/published-documentation#309] [Q&A] [answered]
The sample table files at https://osf.io/rdeks/ ("Samples (tables with 0 or 1 example rows) for all C2M2 TSVs") are very useful or essential in preparing compliant datapackage submissions, but I'm unable to download them except manually one by one. Is there a way to script this? I.e. urls for the raw files, or the bundle?

**Comment by ACharbonneau** [comment, marked as answer]
Select their parent folder and then 'Download as zip'.

![image](https://user-images.githubusercontent.com/1719360/151624511-9e0ff50c-5405-4225-9a0e-cfc0329dd60e.png)

**Reply by jeremyjyang** [reply]
Thanks, very helpful, much better than one by one. However not scriptable I guess.

**Comment by ACharbonneau** [comment]
https://github.com/osfclient/osfclient

**Reply by jeremyjyang** [reply]
Thanks, seems good but apparently I need better OSF authentication. ORCID didn't suffice apparently. However, this works

wget -O - 'https://osf.io/hj9br/download' >anatomy.tsv
etc.

but osfclient will be handy to generate a listing.

**Reply by jeremyjyang** [reply]
I was mistaken, my OSF permissions were fine, just needed to find project ID, username and reset password. Then however, there are no files in "from all storages for a project":

```
(cfde) lengua$ osf -h
usage: osf [-h] [-u USERNAME] [-p PROJECT] [-v]
           {clone,init,fetch,geturl,list,ls,upload,remove,rm} ...

osf is a command-line program to up and download
files from osf.io.

These are common osf commands:

    init      Set up a .osfcli.config file
    clone     Copy all files from all storages of a project
    fetch     Fetch an individual file from a project
    geturl    Get download url for an individual file from a project
    list      List all files from all storages for a project
    upload    Upload a new file to an existing project
    remove    Remove a file from a project's storage

See 'osf <command> -h' to read about a specific command.

positional arguments:
  {clone,init,fetch,geturl,list,ls,upload,remove,rm}

options:
  -h, --help            show this help message and exit
  -u USERNAME, --username USERNAME
                        OSF username. Provide your password via OSF_PASSWORD
                        environment variable
  -p PROJECT, --project PROJECT
                        OSF project ID
  -v, --version         show program's version number and exit
(cfde) lengua$ export OSF_PASSWORD="*******"
(cfde) lengua$ osf -p c8txv -u jeremyjyang@gmail.com list
```
_(Works with no error, but no output.)_

**Reply by ACharbonneau** [reply]
hmm. Luckily one of the people that wrote that is around here somewhere :) @ctb do you have any idea what's going wrong here?

**Comment by ctb** [comment]
this is going to be a bad answer, but: osfclient isn't really maintained, and I have shuffled off that project :(.

**Reply by ctb** [reply]
the zipfile download should be scriptable via download link and you can use `unzip` at the command line to list files, and `unzip -p` to print the contents to stdout.
